src/jarabe/journal/listview.py

```diff
@@ -711,6 +716,17 @@ def __drag_data_get_cb(self, widget, context, selection, info, time):
         model = self.tree_view.get_model()
         model.do_drag_data_get(path, selection)
 
+    def __button_press_event_cb(self, widget, event):
```

Comment:
On Mon, Feb 22, 2016, 18:32 Abhijit Patel <notifications@github.com> wrote:

In src <https://github.com/sugarlabs/sugar/pull/656#discussion_r53593069>/
<https://github.com/sugarlabs/sugar/pull/656#discussion_r53593069>jarabe
<https://github.com/sugarlabs/sugar/pull/656#discussion_r53593069>/journal/
<https://github.com/sugarlabs/sugar/pull/656#discussion_r53593069>
listview.py
<https://github.com/sugarlabs/sugar/pull/656#discussion_r53593069>:

> @@ -711,6 +716,17 @@ def __drag_data_get_cb(self, widget, context,
selection, info, time): > model = self.tree_view.get_model() >
model.do_drag_data_get(path, selection) > > + def
__button_press_event_cb(self, widget, event):

The previously used tree_view.enable_model_drag_source returned the whole
row to be drag. It didn't need any icon or something, the whole row would
be seen lifted.

But as for using the tree_view.drag_source_set sets up a widget on which
drag operation will start but icon wont be set automatically

For starting the drag operation here we are using the button-press-event.

So I have created another function for the button-press-event and also to
set up an icon for indicating the drag.


Yes, but is this the best signal?  Isn't this called every time the user
interacts with the tree view, even if they are just clicking to start an
activity or something?  Can we change it so that we don't load an icon
(expensive operation?) every button event?

—
Reply to this email directly or view it on
<https://github.com/sugarlabs/sugar/pull/656/files#r53593069>GitHub
<https://github.com/sugarlabs/sugar/pull/656/files#r53593069>.<img src="
https://ci4.googleusercontent.com/proxy/ijBkf6ev6uXfwIuPsOxS2GEUGfKbBVpmg7znMfveqOvPSryfOjVF0od8ZvC4zXipQ-M0VMJSIQVNodoYeJOxoIzwCqlGl5LkeRQ1gqxDiwnbZ_funNhB16LBOV4SUMTUjqeWkyMR3xWkJyYKow1QK-2J6k2AiQ=s0-d-e1-ft#https://github.com/notifications/beacon/AFvjmmI9Ozl0iITo8Ul8PRjC3QLAxa5eks5pmrDwgaJpZM4He_IM.gif
">